tests/unit/test_addtoken.py

```diff
@@ -0,0 +1,15 @@
+from unittest import mock, TestCase
+from add_token import print_short
+
+class Test_addtoken(TestCase):
+    @mock.patch('builtins.print')
+    @mock.patch('add_token.get_network_info', return_value=['10.23.53.54', ['10.23.53.54'], '32'])
+    def test_single(self, single, mock_print):
+        print_short('t', 'c')
+        mock_print.assert_called_with('microk8s join 10.23.53.54:32/t/c')
+    
+    @mock.patch('builtins.print')
+    @mock.patch('add_token.get_network_info', return_value=['d_ip', ['ip1', 'ip2', 'd_ip'], '4'])
+    def test_multiple(self, multiple, mock_print):
+        print_short('t', 'c')
+        mock_print.assert_called_with('microk8s join ip2:4/t/c')
```

Comment:
this works, but does not tell the whole story. please make sure that you test all print invocations are as expected, not only the last one.

Reply:
Sure!